src/sentry/static/sentry/app/actionCreators/globalSelection.tsx

```diff
@@ -1,12 +1,69 @@
+import * as ReactRouter from 'react-router';
+import * as Sentry from '@sentry/browser';
 import isInteger from 'lodash/isInteger';
 import omit from 'lodash/omit';
 import qs from 'query-string';
-import * as Sentry from '@sentry/browser';
 
+import {Environment} from 'app/types';
 import {defined} from 'app/utils';
 import {getUtcDateString} from 'app/utils/dates';
 import GlobalSelectionActions from 'app/actions/globalSelectionActions';
 
+/**
+ * Note this is the internal project.id, NOT the slug, but it is the stringified version of it
+ */
+type ProjectId = string | number;
+type EnvironmentId = Environment['id'];
+
+type Options = {
+  /**
+   * List of parameters to remove when changing URL params
+   */
+  resetParams?: string[];
+};
+
+/**
+ * Can be relative time string or absolute (using start and end dates)
+ */
+type DateTimeObject = {
+  start?: Date | string | null;
+  end?: Date | string | null;
+  statsPeriod?: string | null;
+  utc?: string | boolean | null;
+  /**
+   * @deprecated
+   */
+  period?: string | null;
+};
+
+// Get Params type from `getParams` helper
+// type Params = Parameters<typeof getParams>[0];
+
+/**
+ * Cast project ids to strings, as everything is assumed to be a string in URL params
+ *
+ * Discover v1 uses a different interface, and passes slightly different datatypes e.g. Date for dates
+ */
+// type UrlParams = {project?: string[]; environment?: EnvironmentId[]} & Params;
+type UrlParams = {
+  project?: ProjectId[];
```

Comment:
It might be better to rename it to `projects` as it is an array or to be more precise `projectsId`

Reply:
Renaming will diverge from what the URL parameters are. Having them be consistent means we have to do less translations mentally.

Reply:
Yeah, also not something that should be tackled in this PR as it will probably get hairy